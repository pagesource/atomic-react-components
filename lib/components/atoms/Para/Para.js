// @flow
import React from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';

import styles from './Para.style';

type Props = {
  children: Node,
  className?: string | void,
  inheritedStyles?: string,
  elementLocator?: string,
};

const Para = ({ children, className, inheritedStyles, elementLocator, ...others }: Props): Node => {
  const dataLocator = elementLocator
    ? {
        'data-locator': `para-${elementLocator}`,
      }
    : {};

  return (
    <p {...others} className={className} {...dataLocator}>
      {children}
    </p>
  );
};

Para.defaultProps = {
  inheritedStyles: '',
  className: '',
  elementLocator: '',
};

const StyledPara: ComponentType<Props> = styled(Para)`
  ${styles};
`;

export default StyledPara;

export { Para as ParaVanilla };
