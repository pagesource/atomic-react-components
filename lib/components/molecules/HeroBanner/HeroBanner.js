// @flow
/**
 *
 * HeroBanner
 *
 */
import React from 'react';
import type { Node, ComponentType } from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import styles from './HeroBanner.style';
import type { Props } from './types';

const HeroBanner = ({ className, children, renderMediaContent, variation }: Props): Node => (
  <section className={classnames(className, variation)}>
    <header className="wrap" role="banner">
      {renderMediaContent && <div className="media-wrap">{renderMediaContent()}</div>}
      {children && <div className="hero-content">{children}</div>}
    </header>
  </section>
);

HeroBanner.defaultProps = {
  className: 'hero',
  variation: 'stacked',
};

const StyledHeroBanner: ComponentType<Props> = styled(HeroBanner)`
  ${styles};
`;

export default StyledHeroBanner;

export { HeroBanner as HeroBannerVanilla };
