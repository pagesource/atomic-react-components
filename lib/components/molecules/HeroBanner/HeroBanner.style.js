import { css } from 'styled-components';

const styles = css`
  .wrap {
    ${(props) => (props.variation === 'side-by-side' ? 'display: flex; flex: 1 1 auto;' : '')};
    ${(props) => (props.variation === 'positioned' ? 'position: relative' : '')};
  }

  .hero-content {
    ${(props) => props.theme.colors.HERO_CONTENT};
    text-align: center;
    ${(props) =>
      props.variation === 'positioned' ? 'position: absolute; top: 30%; left: 25%' : ''};
    padding: 20px;
  }

  ${(props) => (props.inheritedStyles ? props.inheritedStyles : '')};
`;

export default styles;
