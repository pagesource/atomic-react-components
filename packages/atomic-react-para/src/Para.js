import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import styles from './Para.style';

const Para = ({ children, className, inheritedStyles, elementLocator, ...others }) => {
  const dataLocator = elementLocator
    ? {
        'data-locator': `para-${elementLocator}`,
      }
    : {};

  return (
    <p {...others} className={className} {...dataLocator}>
      {children}
    </p>
  );
};

Para.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
  inheritedStyles: PropTypes.string,
  elementLocator: PropTypes.string,
};

Para.defaultProps = {
  inheritedStyles: '',
  className: '',
  elementLocator: '',
};

const StyledPara = styled(Para)`
  ${styles};
`;

export default StyledPara;

export { Para as ParaVanilla };
