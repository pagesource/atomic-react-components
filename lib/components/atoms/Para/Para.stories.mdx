import { Meta, Story, Preview, Props  } from '@storybook/addon-docs/blocks';
import Para, { ParaVanilla } from '.';

<Meta title="Atoms/Para" component={ParaVanilla} />


### Knobs
<Preview>
  <Story name="Knobs">
<ParaVanilla>This is plain text</ParaVanilla>
  </Story>
</Preview>

### Default
<Preview>
  <Story name="Default">
<Para elementLocator="my-text">This is plain text with element locator.</Para>
  </Story>
</Preview>
