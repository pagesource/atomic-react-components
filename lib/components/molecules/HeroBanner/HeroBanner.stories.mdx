import { Meta, Story, Preview, Props } from '@storybook/addon-docs/blocks';
import {
  stackedHeroBanner,
  sideBySideHeroBanner,
  positionedExternalContent,
} from './HeroBanner.mock';
import Button from '../../atoms/Button/Button';
import primaryButton from '../../atoms/Button/Button.mock';
import Para from '../../atoms/Para/Para';
import HeroBanner, {HeroBannerVanilla} from '.';

<Meta title="Molecules/Hero Banner" component={HeroBannerVanilla} />

### Hero Banner
<Preview>
  <Story name="Hero Banner">
  <HeroBanner {...stackedHeroBanner}>
      <Para>Content will be displayed here in the banner</Para>
    </HeroBanner>
  </Story>
</Preview>

### Stacked
<Preview>
  <Story name="Stacked">
<HeroBanner {...stackedHeroBanner}>
      <Para>Content will be displayed here in the banner</Para>
      <Button {...primaryButton}>CTA</Button>
    </HeroBanner>
  </Story>
</Preview>

### Side By Side
<Preview>
  <Story name="Input Type Radio">
 <HeroBanner {...sideBySideHeroBanner}>
      <Para>Content will be displayed here in the banner</Para>
      <Button {...primaryButton}>CTA</Button>
    </HeroBanner>
  </Story>
</Preview>

### Positioned Content
<Preview>
  <Story name="Positioned Content">
 <HeroBanner {...positionedExternalContent}>
      <Para>Content will be displayed here in the banner</Para>
      <Button {...primaryButton}>CTA</Button>
    </HeroBanner>
  </Story>
</Preview>

## Props ðŸ”§

<Props of={HeroBannerVanilla} />

