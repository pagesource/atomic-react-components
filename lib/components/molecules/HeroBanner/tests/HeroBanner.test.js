import React from 'react';
import { shallowWithTheme } from '../../../../../__mock__/themeMock';
import { HeroBannerVanilla } from '../index';
import { stackedHeroBanner } from '../HeroBanner.mock';
import Para from '../../../atoms/Para/Para';

describe('<HeroBanner />', () => {
  let HeroBannerComponent = '';
  beforeEach(() => {
    HeroBannerComponent = shallowWithTheme(<HeroBannerVanilla {...stackedHeroBanner} />);
  });

  test('should render correctly', () => {
    expect(HeroBannerComponent).toMatchSnapshot();
  });

  test('should render a div', () => {
    expect(HeroBannerComponent.find('section').length).toEqual(1);
  });

  test('should render a header', () => {
    expect(HeroBannerComponent.find('.wrap').length).toEqual(1);
  });

  test('should render a header with role as banner', () => {
    expect(HeroBannerComponent.find('header').prop('role')).toEqual('banner');
  });

  test.skip('should render Hero Image', () => {
    expect(HeroBannerComponent.find('Image').length).toEqual(1);
  });

  test.skip('should render children', () => {
    HeroBannerComponent = shallowWithTheme(
      <HeroBannerVanilla {...stackedHeroBanner}>
        <Para>Content of the banner will be placed here</Para>
      </HeroBannerVanilla>
    );
    expect(HeroBannerComponent.find('Para').length).toEqual(1);
  });
});
