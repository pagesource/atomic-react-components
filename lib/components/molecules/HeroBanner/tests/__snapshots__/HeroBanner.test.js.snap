// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeroBanner /> should render correctly 1`] = `
<section
  className="hero stacked"
>
  <header
    className="wrap"
    role="banner"
  >
    <div
      className="media-wrap"
    >
      <Image__StyledImage
        alt="Dummy Image"
        placeholderSrc="/images/image-placeholder.png"
        src="/images/bg.jpg"
      />
    </div>
  </header>
</section>
`;
